### DISCUSSION TITLE
Feature Request: sas7bdat reader

### DISCUSSION THREAD

**Discussion by alanpaulkwan** [duckdb/duckdb#13381] [Ideas]
I believe I raised this question once, and I was told it's impossible without a c++ plugin. 

https://github.com/olivia76/cpp-sas7bdat

SAS is how the old world stores data, is it possible to integrate this library to read sas files?

**Comment by rustyconover** [comment]
Seems like it would be possible to do, but SAS isn't a very popular format so it is unlikely it would just happen without sponsorship.

**Reply by alanpaulkwan** [reply]
If I may, counterargument would be: academics teach students tools, DuckDB could be the classroom default as academics have the misfortune of dealing with sas7bdat a lot

**Reply by dylanmeysmans** [reply]
> Seems like it would be possible to do, but SAS isn't a very popular format so it is unlikely it would just happen without sponsorship.

@rustyconover Thank you for clarifying that! Could it be that SAS and its `.sas7bdat` files are actually quite commonly used in certain heavily regulated industries, such as finance and healthcare, as per https://github.com/duckdb/duckdb/discussions/10801? If so, could this relatively minor addition make these large industries much more likely to adopt DuckDB instead of SAS, in view of licensing costs for SAS?

**Comment by dylanmeysmans** [comment]
> I believe I raised this question once, and I was told it's impossible without a c++ plugin.
> 
> https://github.com/olivia76/cpp-sas7bdat
> 
> SAS is how the old world stores data, is it possible to integrate this library to read sas files?

@alanpaulkwan I managed to implement this as a community extension, see https://github.com/mettekou/duckdb-read-stat.
